// @govtechsg/token-registry/src/contracts/factories/CalculateTradeTrustERC721Selector__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Signer } from "ethers";
import { Provider, TransactionRequest } from "@ethersproject/providers";
import { Contract, ContractFactory, Overrides } from "@ethersproject/contracts";

import type { CalculateTradeTrustERC721Selector } from "../CalculateTradeTrustERC721Selector";

export class CalculateTradeTrustERC721Selector__factory extends ContractFactory {
  constructor(signer?: Signer) {
    super(_abi, _bytecode, signer);
  }

  deploy(overrides?: Overrides): Promise<CalculateTradeTrustERC721Selector> {
    return super.deploy(
      overrides || {}
    ) as Promise<CalculateTradeTrustERC721Selector>;
  }
  getDeployTransaction(overrides?: Overrides): TransactionRequest {
    return super.getDeployTransaction(overrides || {});
  }
  attach(address: string): CalculateTradeTrustERC721Selector {
    return super.attach(address) as CalculateTradeTrustERC721Selector;
  }
  connect(signer: Signer): CalculateTradeTrustERC721Selector__factory {
    return super.connect(signer) as CalculateTradeTrustERC721Selector__factory;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): CalculateTradeTrustERC721Selector {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as CalculateTradeTrustERC721Selector;
  }
}

const _abi = [
  {
    constant: true,
    inputs: [],
    name: "calculateSelector",
    outputs: [
      {
        internalType: "bytes4",
        name: "",
        type: "bytes4",
      },
    ],
    payable: false,
    stateMutability: "pure",
    type: "function",
  },
];

const _bytecode =
  "0x6080604052348015600f57600080fd5b5060908061001e6000396000f3fe6080604052348015600f57600080fd5b506004361060285760003560e01c8063bb71eb3b14602d575b600080fd5b60336050565b604080516001600160e01b03199092168252519081900360200190f35b639f9e69f360e01b9056fea265627a7a7231582081342d14eda7c674724dd72becbafd0fbb1c27d94c13890b5ff1065023d2ac3364736f6c63430005100032";
