// @govtechsg/token-registry/src/contracts/factories/ITitleEscrowCreator__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer } from "ethers";
import { Provider } from "@ethersproject/providers";

import type { ITitleEscrowCreator } from "../ITitleEscrowCreator";

export class ITitleEscrowCreator__factory {
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): ITitleEscrowCreator {
    return new Contract(address, _abi, signerOrProvider) as ITitleEscrowCreator;
  }
}

const _abi = [
  {
    constant: false,
    inputs: [
      {
        internalType: "address",
        name: "tokenRegistry",
        type: "address",
      },
      {
        internalType: "address",
        name: "beneficiary",
        type: "address",
      },
      {
        internalType: "address",
        name: "holder",
        type: "address",
      },
    ],
    name: "deployNewTitleEscrow",
    outputs: [
      {
        internalType: "address",
        name: "",
        type: "address",
      },
    ],
    payable: false,
    stateMutability: "nonpayable",
    type: "function",
  },
];
