// @govtechsg/document-store/dist/contracts/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "BaseDocumentStore__factory", {
  enumerable: true,
  get: function get() {
    return _BaseDocumentStore__factory.BaseDocumentStore__factory;
  }
});
Object.defineProperty(exports, "BasePaymaster__factory", {
  enumerable: true,
  get: function get() {
    return _BasePaymaster__factory.BasePaymaster__factory;
  }
});
Object.defineProperty(exports, "BaseRelayRecipient__factory", {
  enumerable: true,
  get: function get() {
    return _BaseRelayRecipient__factory.BaseRelayRecipient__factory;
  }
});
Object.defineProperty(exports, "CalculateGsnCapableSelector__factory", {
  enumerable: true,
  get: function get() {
    return _CalculateGsnCapableSelector__factory.CalculateGsnCapableSelector__factory;
  }
});
Object.defineProperty(exports, "ConfigurableTrustForwarder__factory", {
  enumerable: true,
  get: function get() {
    return _ConfigurableTrustForwarder__factory.ConfigurableTrustForwarder__factory;
  }
});
Object.defineProperty(exports, "DocumentStore__factory", {
  enumerable: true,
  get: function get() {
    return _DocumentStore__factory.DocumentStore__factory;
  }
});
Object.defineProperty(exports, "DocumentStoreCreator__factory", {
  enumerable: true,
  get: function get() {
    return _DocumentStoreCreator__factory.DocumentStoreCreator__factory;
  }
});
Object.defineProperty(exports, "DocumentStoreWithRevokeReasons__factory", {
  enumerable: true,
  get: function get() {
    return _DocumentStoreWithRevokeReasons__factory.DocumentStoreWithRevokeReasons__factory;
  }
});
Object.defineProperty(exports, "ERC165__factory", {
  enumerable: true,
  get: function get() {
    return _ERC165__factory.ERC165__factory;
  }
});
Object.defineProperty(exports, "Forwarder__factory", {
  enumerable: true,
  get: function get() {
    return _Forwarder__factory.Forwarder__factory;
  }
});
Object.defineProperty(exports, "GsnCapable__factory", {
  enumerable: true,
  get: function get() {
    return _GsnCapable__factory.GsnCapable__factory;
  }
});
Object.defineProperty(exports, "GsnCapableDocumentStore__factory", {
  enumerable: true,
  get: function get() {
    return _GsnCapableDocumentStore__factory.GsnCapableDocumentStore__factory;
  }
});
Object.defineProperty(exports, "GsnEip712Library__factory", {
  enumerable: true,
  get: function get() {
    return _GsnEip712Library__factory.GsnEip712Library__factory;
  }
});
Object.defineProperty(exports, "IERC165__factory", {
  enumerable: true,
  get: function get() {
    return _IERC165__factory.IERC165__factory;
  }
});
Object.defineProperty(exports, "IForwarder__factory", {
  enumerable: true,
  get: function get() {
    return _IForwarder__factory.IForwarder__factory;
  }
});
Object.defineProperty(exports, "IKnowForwarderAddress__factory", {
  enumerable: true,
  get: function get() {
    return _IKnowForwarderAddress__factory.IKnowForwarderAddress__factory;
  }
});
Object.defineProperty(exports, "IPaymaster__factory", {
  enumerable: true,
  get: function get() {
    return _IPaymaster__factory.IPaymaster__factory;
  }
});
Object.defineProperty(exports, "IRelayHub__factory", {
  enumerable: true,
  get: function get() {
    return _IRelayHub__factory.IRelayHub__factory;
  }
});
Object.defineProperty(exports, "IRelayRecipient__factory", {
  enumerable: true,
  get: function get() {
    return _IRelayRecipient__factory.IRelayRecipient__factory;
  }
});
Object.defineProperty(exports, "IStakeManager__factory", {
  enumerable: true,
  get: function get() {
    return _IStakeManager__factory.IStakeManager__factory;
  }
});
Object.defineProperty(exports, "NaivePaymaster__factory", {
  enumerable: true,
  get: function get() {
    return _NaivePaymaster__factory.NaivePaymaster__factory;
  }
});
Object.defineProperty(exports, "Ownable__factory", {
  enumerable: true,
  get: function get() {
    return _Ownable__factory.Ownable__factory;
  }
});
Object.defineProperty(exports, "OwnableDocumentStore__factory", {
  enumerable: true,
  get: function get() {
    return _OwnableDocumentStore__factory.OwnableDocumentStore__factory;
  }
});
Object.defineProperty(exports, "OwnableUpgradeable__factory", {
  enumerable: true,
  get: function get() {
    return _OwnableUpgradeable__factory.OwnableUpgradeable__factory;
  }
});
Object.defineProperty(exports, "UpgradableDocumentStore__factory", {
  enumerable: true,
  get: function get() {
    return _UpgradableDocumentStore__factory.UpgradableDocumentStore__factory;
  }
});

var _BaseDocumentStore__factory = require("./factories/BaseDocumentStore__factory");

var _BasePaymaster__factory = require("./factories/BasePaymaster__factory");

var _BaseRelayRecipient__factory = require("./factories/BaseRelayRecipient__factory");

var _CalculateGsnCapableSelector__factory = require("./factories/CalculateGsnCapableSelector__factory");

var _ConfigurableTrustForwarder__factory = require("./factories/ConfigurableTrustForwarder__factory");

var _DocumentStore__factory = require("./factories/DocumentStore__factory");

var _DocumentStoreCreator__factory = require("./factories/DocumentStoreCreator__factory");

var _DocumentStoreWithRevokeReasons__factory = require("./factories/DocumentStoreWithRevokeReasons__factory");

var _ERC165__factory = require("./factories/ERC165__factory");

var _Forwarder__factory = require("./factories/Forwarder__factory");

var _GsnCapable__factory = require("./factories/GsnCapable__factory");

var _GsnCapableDocumentStore__factory = require("./factories/GsnCapableDocumentStore__factory");

var _GsnEip712Library__factory = require("./factories/GsnEip712Library__factory");

var _IERC165__factory = require("./factories/IERC165__factory");

var _IForwarder__factory = require("./factories/IForwarder__factory");

var _IKnowForwarderAddress__factory = require("./factories/IKnowForwarderAddress__factory");

var _IPaymaster__factory = require("./factories/IPaymaster__factory");

var _IRelayHub__factory = require("./factories/IRelayHub__factory");

var _IRelayRecipient__factory = require("./factories/IRelayRecipient__factory");

var _IStakeManager__factory = require("./factories/IStakeManager__factory");

var _NaivePaymaster__factory = require("./factories/NaivePaymaster__factory");

var _Ownable__factory = require("./factories/Ownable__factory");

var _OwnableDocumentStore__factory = require("./factories/OwnableDocumentStore__factory");

var _OwnableUpgradeable__factory = require("./factories/OwnableUpgradeable__factory");

var _UpgradableDocumentStore__factory = require("./factories/UpgradableDocumentStore__factory");
