// @govtechsg/token-registry/dist/contracts/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "CalculateSelector__factory", {
  enumerable: true,
  get: function get() {
    return _CalculateSelector__factory.CalculateSelector__factory;
  }
});
Object.defineProperty(exports, "CalculateTradeTrustERC721Selector__factory", {
  enumerable: true,
  get: function get() {
    return _CalculateTradeTrustERC721Selector__factory.CalculateTradeTrustERC721Selector__factory;
  }
});
Object.defineProperty(exports, "Context__factory", {
  enumerable: true,
  get: function get() {
    return _Context__factory.Context__factory;
  }
});
Object.defineProperty(exports, "ERC165__factory", {
  enumerable: true,
  get: function get() {
    return _ERC165__factory.ERC165__factory;
  }
});
Object.defineProperty(exports, "ERC721__factory", {
  enumerable: true,
  get: function get() {
    return _ERC721__factory.ERC721__factory;
  }
});
Object.defineProperty(exports, "ERC721Enumerable__factory", {
  enumerable: true,
  get: function get() {
    return _ERC721Enumerable__factory.ERC721Enumerable__factory;
  }
});
Object.defineProperty(exports, "ERC721Full__factory", {
  enumerable: true,
  get: function get() {
    return _ERC721Full__factory.ERC721Full__factory;
  }
});
Object.defineProperty(exports, "ERC721Metadata__factory", {
  enumerable: true,
  get: function get() {
    return _ERC721Metadata__factory.ERC721Metadata__factory;
  }
});
Object.defineProperty(exports, "ERC721Mintable__factory", {
  enumerable: true,
  get: function get() {
    return _ERC721Mintable__factory.ERC721Mintable__factory;
  }
});
Object.defineProperty(exports, "ERC721MintableFull__factory", {
  enumerable: true,
  get: function get() {
    return _ERC721MintableFull__factory.ERC721MintableFull__factory;
  }
});
Object.defineProperty(exports, "HasHolder__factory", {
  enumerable: true,
  get: function get() {
    return _HasHolder__factory.HasHolder__factory;
  }
});
Object.defineProperty(exports, "HasNamedBeneficiary__factory", {
  enumerable: true,
  get: function get() {
    return _HasNamedBeneficiary__factory.HasNamedBeneficiary__factory;
  }
});
Object.defineProperty(exports, "IERC165__factory", {
  enumerable: true,
  get: function get() {
    return _IERC165__factory.IERC165__factory;
  }
});
Object.defineProperty(exports, "IERC721__factory", {
  enumerable: true,
  get: function get() {
    return _IERC721__factory.IERC721__factory;
  }
});
Object.defineProperty(exports, "IERC721Enumerable__factory", {
  enumerable: true,
  get: function get() {
    return _IERC721Enumerable__factory.IERC721Enumerable__factory;
  }
});
Object.defineProperty(exports, "IERC721Metadata__factory", {
  enumerable: true,
  get: function get() {
    return _IERC721Metadata__factory.IERC721Metadata__factory;
  }
});
Object.defineProperty(exports, "IERC721Receiver__factory", {
  enumerable: true,
  get: function get() {
    return _IERC721Receiver__factory.IERC721Receiver__factory;
  }
});
Object.defineProperty(exports, "ITitleEscrow__factory", {
  enumerable: true,
  get: function get() {
    return _ITitleEscrow__factory.ITitleEscrow__factory;
  }
});
Object.defineProperty(exports, "ITitleEscrowCreator__factory", {
  enumerable: true,
  get: function get() {
    return _ITitleEscrowCreator__factory.ITitleEscrowCreator__factory;
  }
});
Object.defineProperty(exports, "MinterRole__factory", {
  enumerable: true,
  get: function get() {
    return _MinterRole__factory.MinterRole__factory;
  }
});
Object.defineProperty(exports, "TitleEscrow__factory", {
  enumerable: true,
  get: function get() {
    return _TitleEscrow__factory.TitleEscrow__factory;
  }
});
Object.defineProperty(exports, "TitleEscrowCreator__factory", {
  enumerable: true,
  get: function get() {
    return _TitleEscrowCreator__factory.TitleEscrowCreator__factory;
  }
});
Object.defineProperty(exports, "TradeTrustERC721__factory", {
  enumerable: true,
  get: function get() {
    return _TradeTrustERC721__factory.TradeTrustERC721__factory;
  }
});

var _CalculateSelector__factory = require("./factories/CalculateSelector__factory");

var _CalculateTradeTrustERC721Selector__factory = require("./factories/CalculateTradeTrustERC721Selector__factory");

var _Context__factory = require("./factories/Context__factory");

var _ERC165__factory = require("./factories/ERC165__factory");

var _ERC721__factory = require("./factories/ERC721__factory");

var _ERC721Enumerable__factory = require("./factories/ERC721Enumerable__factory");

var _ERC721Full__factory = require("./factories/ERC721Full__factory");

var _ERC721Metadata__factory = require("./factories/ERC721Metadata__factory");

var _ERC721Mintable__factory = require("./factories/ERC721Mintable__factory");

var _ERC721MintableFull__factory = require("./factories/ERC721MintableFull__factory");

var _HasHolder__factory = require("./factories/HasHolder__factory");

var _HasNamedBeneficiary__factory = require("./factories/HasNamedBeneficiary__factory");

var _IERC165__factory = require("./factories/IERC165__factory");

var _IERC721__factory = require("./factories/IERC721__factory");

var _IERC721Enumerable__factory = require("./factories/IERC721Enumerable__factory");

var _IERC721Metadata__factory = require("./factories/IERC721Metadata__factory");

var _IERC721Receiver__factory = require("./factories/IERC721Receiver__factory");

var _ITitleEscrow__factory = require("./factories/ITitleEscrow__factory");

var _ITitleEscrowCreator__factory = require("./factories/ITitleEscrowCreator__factory");

var _MinterRole__factory = require("./factories/MinterRole__factory");

var _TitleEscrow__factory = require("./factories/TitleEscrow__factory");

var _TitleEscrowCreator__factory = require("./factories/TitleEscrowCreator__factory");

var _TradeTrustERC721__factory = require("./factories/TradeTrustERC721__factory");
