// @govtechsg/token-registry/dist/contracts/factories/HasHolder__factory.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HasHolder__factory = void 0;

var _ethers = require("ethers");

/* Autogenerated file. Do not edit manually. */

/* tslint:disable */

/* eslint-disable */
class HasHolder__factory {
  static connect(address, signerOrProvider) {
    return new _ethers.Contract(address, _abi, signerOrProvider);
  }

}

exports.HasHolder__factory = HasHolder__factory;
const _abi = [{
  inputs: [{
    internalType: "address",
    name: "_holder",
    type: "address"
  }],
  payable: false,
  stateMutability: "nonpayable",
  type: "constructor"
}, {
  anonymous: false,
  inputs: [{
    indexed: true,
    internalType: "address",
    name: "previousHolder",
    type: "address"
  }, {
    indexed: true,
    internalType: "address",
    name: "newHolder",
    type: "address"
  }],
  name: "HolderChanged",
  type: "event"
}, {
  constant: true,
  inputs: [],
  name: "holder",
  outputs: [{
    internalType: "address",
    name: "",
    type: "address"
  }],
  payable: false,
  stateMutability: "view",
  type: "function"
}];
