// @govtechsg/document-store/dist/contracts/factories/BaseRelayRecipient__factory.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BaseRelayRecipient__factory = void 0;

var _ethers = require("ethers");

/* Autogenerated file. Do not edit manually. */

/* tslint:disable */

/* eslint-disable */
class BaseRelayRecipient__factory {
  static connect(address, signerOrProvider) {
    return new _ethers.Contract(address, _abi, signerOrProvider);
  }

}

exports.BaseRelayRecipient__factory = BaseRelayRecipient__factory;
const _abi = [{
  inputs: [],
  name: "trustedForwarder",
  outputs: [{
    internalType: "address",
    name: "",
    type: "address"
  }],
  stateMutability: "view",
  type: "function"
}, {
  inputs: [],
  name: "versionRecipient",
  outputs: [{
    internalType: "string",
    name: "",
    type: "string"
  }],
  stateMutability: "view",
  type: "function"
}, {
  inputs: [{
    internalType: "address",
    name: "forwarder",
    type: "address"
  }],
  name: "isTrustedForwarder",
  outputs: [{
    internalType: "bool",
    name: "",
    type: "bool"
  }],
  stateMutability: "view",
  type: "function"
}];
